### ISSUE TITLE
Missing files while building VS Solution

### ISSUE THREAD

**Issue by Silloty1** [microsoft/MixedReality-WebRTC#206]
Hi

I'm stuck at the step : Building From Sources : https://microsoft.github.io/MixedReality-WebRTC/manual/building.html

After cloning the repository from GitHub and trying to build the solution, two errors occur :
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3030	Could not copy the file "C:\mr-webrtc\bin\Win32\x64\Debug\Microsoft.MixedReality.WebRTC.Native.pdb" because it was not found.	Microsoft.MixedReality.WebRTC.Tests	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets	4570	

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3030	Could not copy the file "C:\mr-webrtc\bin\Win32\x64\Debug\Microsoft.MixedReality.WebRTC.Native.dll" because it was not found.	Microsoft.MixedReality.WebRTC.Tests	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets	4570	

I understand that files are missing, but I did all the steps before correctly.

Any suggestion ?

Thank you

**Comment by djee-ms** [comment]
The files are compiled by the Visual Studio solution. If they are missing, it means the corresponding project was not compiled correctly.

**Comment by Silloty1** [comment]
I've opened the Visual Studio solution on the folder C:\mr-webrtc\Microsoft.MixedReality.WebRTC.sln.
I've opened this one right after finishing cloning the repo from GitHub via GitBash

**Comment by djee-ms** [comment]
Yes but then you need to compile the projects in the solution. And if you did then there should have been compile errors.

**Comment by Silloty1** [comment]
Hi

Thanks for the reply
After succeeding building the projects : 
Microsoft.MixedReality.WebRTC
Microsoft.MixedReality.WebRTC.Native.UWP

I have an error while building Microsoft.MixedReality.WebRTC.Native.Win32

The error is the following : Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK1104	cannot open file 'webrtc.lib'	Microsoft.MixedReality.WebRTC.Native.Win32	C:\mr-webrtc\libs\Microsoft.MixedReality.WebRTC.Native\src\win32\LINK	1

Should I get it on NuGet ?

**Comment by djee-ms** [comment]
You shouldn't get it **on** NuGet(.org) but rather **via** NuGet package restore from inside Visual Studio. It looks like the packages were not restored correctly. Depending on your Visual Studio setup, this is either done automatically or you have to do it manually, for example by right-clicking on the project and selecting "Manage NuGet Packages...". To confirm the packages are restored correctly, you can also check the content of the `<root>/packages/` folder.

**Comment by Silloty1** [comment]
What should I get from the package manager then ? I didn't get it

**Comment by djee-ms** [comment]
Sorry I didn't get a notification for your reply. When compiling with the Visual Studio solution, the NuGet package manager will restore the NuGet packages for the core implementation, which contain precompiled versions of `webrtc.lib` from the Google repository. Those are then linked against our implementation `Microsoft.MixedReality.WebRTC.Native.dll`, which is built by the project with the same name (one for Desktop, one for UWP). Then that DLL is copied to the Unity project.

Currently if you're missing `webrtc.lib` then it means the restore of `Microsoft.MixedReality.WebRTC.Native.Core.*` packages was not done.

![image](https://user-images.githubusercontent.com/49064175/76614291-aa0f3d00-6517-11ea-8503-f181ba472dff.png)

You can see in that screenshot that I don't have the latest packages (1.0.3) but I have the version before (1.0.2) so `webrtc.lib` is available in `<repo_root>\packages\Microsoft.MixedReality.WebRTC.Native.Core.Desktop.x64.Release.1.0.3\lib\native\lib\win10\x64\release` for example for the Desktop x64 Release config. The content of the `packages` folder is populated by restoring the NuGet packages.

**Comment by Silloty1** [comment]
Hi

So I tried to restore the missing packages as on your screenshot.
Currently, I have this when I'm opening my Nuget Packages Manager :
![image](https://user-images.githubusercontent.com/17072780/76852454-a64b2580-684b-11ea-9c5d-fd6af6495e13.png)

After building again, the same error occurs :/

**Comment by djee-ms** [comment]
Hi @Silloty1, glad that worked! What was the issue in the end? Do you know why it was not working the first time? I would be interested to understand if there is something we can improve, so other users don't have so much trouble!

**Comment by Silloty1** [comment]
Hey !
The issue is that I was doing steps I didn't necessarly needed
I didn't understand completely the tutorial, and I didn't need the C# and C++ installation, but only the Unity Installation. I didn't need the "Building from sources" neither and I just checked the Unity Install and Unity tutorial in the end. I just thought that I needed to complete at least some steps before going to the Unity Tutorial Ahah.
